### PR TITLE
remove JsonBuilder

### PR DESCRIPTION
**Pull request by jaredsburrows**
@ben-manes

### CODE REVIEW

File: src/main/groovy/com/github/benmanes/gradle/versions/reporter/JsonReporter.groovy [hunk 1]
```diff
@@ -1,20 +1,47 @@
...
   @Override
   void write(Appendable printStream, Result result) {
-    printStream.println(new JsonBuilder(result).toPrettyString().stripMargin())
+    JsonAdapter<Result2> jsonAdapter = moshi.adapter(Result2.class)
+    .indent(" ")
+    Result2 result2 = new Result2()
+    result2.count = 1
+    result2.gradle = new GradleUpdateResults()
+//    JsonAdapter<Result> jsonAdapter =
+//      moshi.adapter(Types.newParameterizedType(Result.class, SortedSet.class, com.github.benmanes.gradle.versions.reporter.result.Dependency.class))
```
Comment (ben-manes) [inline src/main/groovy/com/github/benmanes/gradle/versions/reporter/JsonReporter.groovy:40]: delete debug?

**Review by ben-manes — APPROVED** [review]

**Comment by jaredsburrows** [comment]
I pushed this up to save it temporarily. I found more generics and other types that Moshi cannot handle. Looking into this now.

**Comment by jaredsburrows** [comment]
@ben-manes For some reason, changing the types changes `<dependencies>` to `<dependencies class="sorted-set">`.

**Comment by jaredsburrows** [comment]
@ben-manes I was able to fix this via `xstream.aliasSystemAttribute(null, "class")`.

**Comment by ben-manes** [comment]
By switching to `Set` did you lose sorted order? If so, please add a TreeSet copy. We don't want to display the results in hash order as confusing.

**Comment by jaredsburrows** [comment]
@ben-manes Can you explain more here? Moshi cannot handle `SortedSet`, suggested I use `Set`. Which test was checking the order?

**Comment by ben-manes** [comment]
I don’t think we have tests, but the output should show dependencies in sorted order. You can just sort at the usages though via a new TreeSet copy

**Pull request merged**